### DISCUSSION TITLE
Is there a way to show placeholders for snippets that just do `$1`?

### DISCUSSION THREAD

**Discussion by emi2k01** [L3MON4D3/LuaSnip#544] [Q&A]
Sometimes when I use a new snippet I don't know if I'm on the last placeholder before `$0` which sometimes is annoying because once you reach `$0` you can't go back.

Thanks!

**Comment by L3MON4D3** [comment]
Ah, if you enable `history` via `ls.config.setup({history=true})` you'll be able to go back to the previous placeholders (but also other snippets, which might be irritating)

**Reply by emi2k01** [reply]
The problem mostly came because I had `<tab>` set to both indenting and jumping. Now I have 2 bindings and now I don't really care to see the placeholder except it'd still be nice to know where I'll jump to.

**Reply by L3MON4D3** [reply]
Mhmmm, for that: you can use `ext_opts` to add virtual text to all kinds of nodes, but that will only show which lines contain an insertNode, but not the column ([example](https://github.com/L3MON4D3/LuaSnip/wiki/Nice-Configs#hint-node-type-with-virtual-text)).
To actually show the column this way we'll have to wait until anticonceal is implemented/ported from vim (though I could swear I had some hacky script for showing the position of the insertNodes at some time.. Can't find it alas :/)

**Comment by leiserfg** [comment]
One way could be to give a default placeholder to all the `$0` so it's easier to know where are they :thinking:, that could go in the global config.
I'm thinking of 

exit_marker: bool|string  
false:  nothing (as now)
 true:  a default value  :door:  or ¶   
 string:  that string
